operator-common/src/main/java/io/strimzi/operator/common/LoggerWrapper.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+
+package io.strimzi.operator.common;
+
+import org.apache.logging.log4j.Logger;
+
+/**
+ * This class wraps up the Log4j2 loggers. It adds the markers that can be used for log filtering.
+ */
+public class LoggerWrapper {
+    private final Logger log4j2Logger;
+    public LoggerWrapper(Logger log4j2Logger) {
+        this.log4j2Logger = log4j2Logger;
+    }
+    // INFO
+    public void info(String msg, Reconciliation r) {
```

Comment:
Isn't this confusing? It looks like the _marker_ should not be in the _msg_ because the rapper adds it. But then it would be much more logical to have the marker as the first param and then a clean message in _msg_ + the in served objects.  I.e. this:

```
...info("Marker1", "{} + {} = {}", 1, 2, 3)
```

than

```
...info("{} + {} = {}", "Marker" 1, 2, 3)
```

Reply:
@tombentley do you like this format better?

Reply:
@scholzj it's not the marker which is added to the message, but the reconciliation itself, which is how we've always logged messages from contexts where there is a reconciliation going on (well, except for the "resource operators", where we never changed the APIs to take a Reconciliation). I think the `Reconciliation.toString()` contains the same information as the marker, so I don't think we should also log the marker itself.

@sknot-rh I think we should make the `Reconciliation` the 1st parameter for all these methods. That way it's clearer at call sites that is it not one of the interpolated arguments for the format string. E.g. a call site would look like

```java
LOG.debug(reconciliation, "Blah blah: resource={}", resource);
```